Framework/Geometry/src/Instrument.cpp

```diff
@@ -1014,7 +1014,18 @@ void Instrument::saveNexus(::NeXus::File *file,
 
   // Now the parameter map, as a NXnote via its saveNexus method
   if (isParametrized()) {
-    const Geometry::ParameterMap &params = *getParameterMap();
+    Geometry::ParameterMap params(*getParameterMap());
+    // Masking is in DetectorInfo. Insert it into ParameterMap so it is saved
+    // alongside other parameters.
+    if (m_detectorInfo) {
```

Comment:
Is this functionality being encapsulated at the right level here? Wouldn't an `DetectorInfo::writeMasksToParameterMap`  be a better option?